static/js/narrow.js

```diff
@@ -599,6 +599,46 @@ exports.by_recipient = function (target_id, opts) {
     }
 };
 
+// Called by the narrow_to_compose_target hotkey.
+exports.to_compose_target = function () {
+    if (!compose_state.composing()) {
+        return;
+    }
+
+    var opts = {
+        trigger: 'narrow_to_compose_target',
+    };
+
+    if (compose_state.get_message_type() === 'stream') {
+        var stream_name = compose_state.stream_name();
+        var stream_id = stream_data.get_stream_id(stream_name);
+        if (!stream_id) {
+            return;
+        }
+        var topics = topic_data.get_recent_names(stream_id);
+        var operators = [{operator: 'stream', operand: stream_name}];
+        var topic = compose_state.topic();
+        if (topics.indexOf(topic) !== -1) {
+            operators.push({operator: 'topic', operand: topic});
+        }
+        exports.activate(operators, opts);
+        return;
+    }
+
+    if (compose_state.get_message_type() === 'private') {
+        var recipient_string = compose_state.recipient();
+        var emails = util.extract_pm_recipients(recipient_string);
+        var invalid = _.reject(emails, people.is_valid_email_for_compose);
+        // If the recipient string is empty or all recipients are
+        // invalid recipients, narrow to all PMs.
+        if (invalid.length === emails.length) {
```

Comment:
I think it should do this if _any_ recipients are invalid, no?

Reply:
This was to mimic the current state; I noticed that if we specify 2 correct and one incorrect recipient, and send something, it gets sent to the two correct recipients anyway. It only made sense to do the same and try narrowing to those recipents.

Reply:
Yeah, I don't think that current behavior is what we want long-term.  We do need to support the `zephyr` case where the client doesn't have the info to figure out the recipients, but that's the only legitimate case for not erroring when we have invalid input here.